app/src/main/java/link/standen/michael/slideshow/ImageActivity.java

```diff
@@ -541,7 +541,7 @@ private int nextImagePosition(boolean forwards){
 			if (newPosition >= fileList.size()){
 				newPosition = 0;
 			}
-		} while (!testPositionIsImage(newPosition));
+		} while (!testPositionIsImage(newPosition) || !testPositionExists(newPosition));
```

Comment:
Should probably call `testPositionExists` before checking if the item is an image

Reply:
The item being an image is a constant lookup, an object's variable instead of interacting with the file system.